### PR TITLE
Frogger

### PR DESCRIPTION
**Pull request by IhorOzerov**
### Frogger game

[Demo](https://ihorozerov.github.io/Frogger-Game/) |
[Code base](https://github.com/IhorOzerov/Frogger-Game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by IhorOzerov** [comment]
I have completed self-checks and fixed code accordingly. Please, check it.

File: submissions/IhorOzerov/frogger/app.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    this.y = config.PLAYER_CONST.y;
+    this.sprite = config.PLAYER_CONST.sprite;
+};
+const player = new Player;
+
+Player.prototype.update = function () {
+    if (this.y < config.Y_START) {
+        setTimeout(function () {
+            player.x = config.PLAYER_CONST.x;
+            player.y = config.PLAYER_CONST.y;
```
Comment (OleksiyRudenko) [inline submissions/IhorOzerov/frogger/app.js:81]: Class definition should not refer a specific instance of the class.
There are means to refer to properties you actually use. Why this? (not a rhetoric question)

File: submissions/IhorOzerov/frogger/app.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    const enemyHeight = this.y + config.PLAYER_CONST.height;
+    const playerWidth = player.x + config.PLAYER_CONST.width;
+    const playerHeight = player.y + config.PLAYER_CONST.height;
+    
+    if (player.x < enemyWidth &&
+        playerWidth > this.x &&
+        player.y < enemyHeight &&
+        playerHeight > this.y) {
+            player.x = config.PLAYER_CONST.x;
+            player.y = config.PLAYER_CONST.y;
```
Comment (OleksiyRudenko) [inline submissions/IhorOzerov/frogger/app.js:53]: Requirements posted/linked by bot addreess specically issue we have in this fragment.
Please identify the requirement, post it here as a comment and fix code accordingly.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@IhorOzerov good start.
However, not all requirements are met.
If anything is not clear feel free asking questions in Students' chat.

File: submissions/IhorOzerov/frogger/app.js (hunk 1 above)
Comment (IhorOzerov) [reply submissions/IhorOzerov/frogger/app.js:81]: I moved the character with coordinates and chose these two points for myself.

File: submissions/IhorOzerov/frogger/app.js (hunk 2 above)
Comment (IhorOzerov) [reply submissions/IhorOzerov/frogger/app.js:53]: `Classes do not refer to any global variables, like global variable player, which is an instance of Player class (referring to global constants and globals provided by the gaming platform like Resources is OK); Hint: pass instance of a game object (or objects) as an argument to other game objects they need to interact with.`

**Review by OleksiyRudenko — COMMENTED** [review]
Further comments on discussions.

File: submissions/IhorOzerov/frogger/app.js (hunk 2 above)
Comment (IhorOzerov) [reply submissions/IhorOzerov/frogger/app.js]: already fixed it, check please

File: submissions/IhorOzerov/frogger/app.js (hunk 1 above)
Comment (IhorOzerov) [reply submissions/IhorOzerov/frogger/app.js]: i am renamed specific instance, please check.
i am not pretty sure am i understand

Comment (IhorOzerov) [reply submissions/IhorOzerov/frogger/app.js]: am i need to make some constants like a math decision, but not like "i am decided that"?
or there is problem exactly in specific instance?

Comment (OleksiyRudenko) [reply submissions/IhorOzerov/frogger/app.js]: Is "i decided that" equivalent to "i just guessed and it works"?
Calculations provide explanations. As a developer tasked to contribute to a project developed before by someone else (often previous contributors are unreachable as people tend to quit companies) you will want to understand what is going on in the code, why and what is behind numbers in particular.

**Review by OleksiyRudenko — APPROVED** [review]
@IhorOzerov great job!

File: submissions/IhorOzerov/frogger/app.js (hunk 1 above)
Comment (IhorOzerov) [reply submissions/IhorOzerov/frogger/app.js]: Second option will be closer to truth. 
After this fixes i am better see how prototype works and where it similar to class sintax. Thank you!

**Pull request merged**